### PR TITLE
added support for HTML comments #68

### PR DESCRIPTION
**Pull request by whisk**
Issue: #68

### CODE REVIEW

File: elem.go [hunk 1]
```diff
@@ -62,6 +62,16 @@ func (t TextNode) Render() string {
 	return string(t)
 }
 
+type CommentNode string
+
+func (c CommentNode) RenderTo(builder *strings.Builder) {
```
Comment (chasefleming) [inline elem.go:67]: Great implementation of the `CommentNode`! One small suggestion for optimization:

```go
func (c CommentNode) RenderTo(builder *strings.Builder) {
	builder.WriteString("<!-- ")
	builder.WriteString(string(c))
	builder.WriteString(" -->")
}
```

File: elem.go [hunk 2]
```diff
@@ -62,6 +62,16 @@ func (t TextNode) Render() string {
...
+	builder.WriteString("<!-- " + string(c) + " -->")
+}
+
+func (c CommentNode) Render() string {
```
Comment (chasefleming) [inline elem.go:71]: Then we can follow the same pattern as the `Element.Render`:

```go
func (c CommentNode) Render() string {
	var builder strings.Builder
	c.RenderTo(&builder)
	return builder.String()
}
```

**Review by chasefleming — CHANGES_REQUESTED** [review]
Great work! Just two minor notes. Thanks so much for the issue creation and work on this.

File: elem.go (hunk 1 above)
Comment (whisk) [reply elem.go:67]: Thanks for this performance tip. Fixed.

File: elem.go (hunk 2 above)
Comment (whisk) [reply elem.go:73]: Fixed.

**Review by chasefleming — APPROVED** [review]
Looks great, thanks!

**Pull request merged**